WeekFields export and typings

StrayAlien opened the pull request:
exporting WeekFields and ComputedDayOfField from locale package.  Plus, typings for same.  Permits one to do: 

```
import {WeekFields} from "@js-joda/locale";
```

rather than import directly from locale/src folder (as per docs) and then have to juggle your ts/babel/jest/etc config to accomodate that one file ... 

I've added a typing for `EnumMap`.  Unsure if you'd like this in the core typings, but, it is in the locale typings atm.  Let me know if you want that changed.

pithu commented:
Hello @StrayAlien 

thanks for this PR, i guess exporting WeekFields makes sense. I am unsure with ComputedDayOfField, because its not part of the threeten API. What is the use for ?

@phueper WDYT ?

StrayAlien commented:
Thanks @pithu - tbh, it was arbitrary as it was in the same code unit - but ... I see that WeekFields does not actually expose it, so, I have removed the ComputedDayOfField typing.  Keep it easy and also means the EnumMap type can go as well.

Also, in the latest commit above I've removed the typing for OffsetTime.toEpochSeconds().  Hope that is okay also.

pithu commented on packages/locale/src/js-joda-locale.js:
```diff
@@ -5,9 +5,12 @@
-    Locale
+    Locale,
+    WeekFields,
+    ComputedDayOfField
```
Its still exported ?

StrayAlien replied:
Thanks @pithu - tests in and export removed.  Hopefully the tests are of the sort your require.

pithu reviewed (commented):
I would expect a very basic test for WeekFields at `packages/locale/test/typescript_definitions/js-joda-locale-tests.ts` similar to the other typing tests

pithu commented on packages/locale/typings/js-joda-locale.d.ts:
```diff
@@ -10,6 +12,22 @@ export class Locale {
+    public static SUNDAY_START: WeekFields;
+    public static ofLocale(locale: Locale): WeekFields;
+    public static of(firstDayOrLocale: core.DayOfWeek | Locale, minDays?: number): WeekFields;
+    public static ofFirstDayOfWeekMinDays(firstDayOfWeek: DayOfWeek, minimalDaysInFirstWeek: number): WeekFields;
```
LGTM, great work, also the tests! 

I am just wondering if we should keep this consistent and use `core.DayOfWeek` instead of `DayOfWeek`. Would this make the import above obsolete ?

InExtremaRes replied:
I am also in favor of moving the `import * as core from '@js-joda/core'` statement to the top and replace the other import.

InExtremaRes commented on packages/locale/typings/js-joda-locale.d.ts:
```diff
@@ -10,6 +12,22 @@ export class Locale {
+    public static SUNDAY_START: WeekFields;
+    public static ofLocale(locale: Locale): WeekFields;
+    public static of(firstDayOrLocale: core.DayOfWeek | Locale, minDays?: number): WeekFields;
+    public static ofFirstDayOfWeekMinDays(firstDayOfWeek: DayOfWeek, minimalDaysInFirstWeek: number): WeekFields;
```
Some of these methods aren't public and shouldn't be typed, in particualr `ofLocale` and `ofFirstDayOfWeekMinDays`.

Also, I think the `of` method could be typed as an overload instead of using an union type. Right now the type accepts a call with a `Locale` _and_ a `number`, which isn't right.

InExtremaRes reviewed (changes requested):
Hi and thank you for contributing. I left just a couple of comments but it looks good to me overall.

StrayAlien commented:
Thanks for comments. Getting there.  

I've updated to reflect those comments.  Thank you.  I used this as a reference - I hope that it is correct: https://www.threeten.org/threetenbp/apidocs/org/threeten/bp/temporal/WeekFields.html.

InExtremaRes commented:
Just one more little thing. Could you please put the `import` statement at the top of the file? Even if for TypeScript is not an issue in this case, It looks a little weird where it is now. Thank you.

StrayAlien commented:
done

pithu reviewed (approved):
LGTM

InExtremaRes reviewed (approved):
👍
